mixer/cmd/reset.go

```diff
@@ -0,0 +1,268 @@
+		if format, err = ioutil.ReadFile(filename); err != nil {
+			return err
+		}
+	} else {
+		format = []byte(strconv.Itoa(int(currentMixFormatInt)))
+	}
+	b.State.Mix.Format = strings.TrimSpace(string(format))
+
+	// Change upstreamVersion file content to the new current version upstreamver
+	var lastStableMixUpstreamVersion []byte
+	filename := filepath.Join(b.Config.Builder.ServerStateDir, "www", b.MixVer, "upstreamver")
+	if lastStableMixUpstreamVersion, err = ioutil.ReadFile(filename); err != nil {
+		return err
+	}
+	if strings.TrimSpace(string(lastStableMixUpstreamVersion)) != b.UpstreamVer {
+		// Set the upstream version to the previous format's latest version
+		b.UpstreamVer = strings.TrimSpace(string(lastStableMixUpstreamVersion))
+		b.UpstreamVerUint32, err = parseUint32(b.UpstreamVer)
+		if err != nil {
+			return errors.Wrapf(err, "Couldn't parse upstream version")
+		}
+		vFile := filepath.Join(b.Config.Builder.VersionPath, b.UpstreamVerFile)
+		if err = ioutil.WriteFile(vFile, []byte(b.UpstreamVer), 0644); err != nil {
+			return err
+		}
+	}
+
+	// Change upstreamURL file content to the new current version upstreamurl
+	var lastStableMixUupstreamURL []byte
+	filename = filepath.Join(b.Config.Builder.ServerStateDir, "www", b.MixVer, "upstreamurl")
+	if lastStableMixUupstreamURL, err = ioutil.ReadFile(filename); err != nil {
+		return err
+	}
+	if strings.TrimSpace(string(lastStableMixUupstreamURL)) != b.UpstreamURL {
+		// Set the upstream version to the previous format's latest version
+		b.UpstreamURL = strings.TrimSpace(string(lastStableMixUupstreamURL))
+		vFile := filepath.Join(b.Config.Builder.VersionPath, b.UpstreamURLFile)
+		if err = ioutil.WriteFile(vFile, []byte(b.UpstreamURL), 0644); err != nil {
+			return err
+		}
+	}
+
+	// Change mixVersion to point to new mixer version
+	if err := ioutil.WriteFile(filepath.Join(b.Config.Builder.VersionPath, b.MixVerFile), []byte(b.MixVer), 0644); err != nil {
+		return err
+	}
+
+	// Make sure update/image/LAST_VER points to new mixer version
+	if err = ioutil.WriteFile(filepath.Join(b.Config.Builder.ServerStateDir, "image", "LAST_VER"), []byte(b.MixVer), 0644); err != nil {
+		return fmt.Errorf("couldn't update LAST_VER file: %s", err)
+	}
+
+	// Make sure update/image/format#/latest points to new mixer version
+	newFormat := "format" + strconv.Itoa(int(currentMixFormatInt))
+	if err = ioutil.WriteFile(filepath.Join(b.Config.Builder.ServerStateDir, "/www/version/", newFormat, "latest"), []byte(b.MixVer), 0644); err != nil {
+		return fmt.Errorf("couldn't update latest file: %s", err)
+	}
+	//update the sig file for the latest
+
+	// Make sure update/image/latest_version points to new mixer version
+	if err = ioutil.WriteFile(filepath.Join(b.Config.Builder.ServerStateDir, "/www/version/", "latest_version"), []byte(b.MixVer), 0644); err != nil {
+		return fmt.Errorf("couldn't update latest_version file: %s", err)
+	}
+	//update the sig file for the latest_version
```

Comment:
same for this, need to decide if updating the sig is a good idea or not

Reply:
I don't think `mixer reset` should change these. I also don't think `mixer reset` should support the --clean flag because it is easy to corrupt manifests when rolling back and updating manifests at an old version.

I think that `mixer reset` should roll back the relevant versions (mixversion/last_ver/previous_ver). When running `mixer build update` or `mixer build all`, manifests ahead of the current mix should be erased otherwise they will become corrupted.

@reaganlo Thoughts?